src/Helper/Uri.php

```diff
@@ -310,15 +310,40 @@ public function getPass(mixed $default = null): ?StringHelper
 
     /**
      * Returns TRUE if the host is valid.
-     * todo: Implement check for host URI segment
      *
      * @param string|null $host
      * @return boolean
      */
     public function checkHost(string $host = null): bool
     {
         if ($host === null) {
-            $host = $this->host;
+            $host = $this->getHost();
+        }
+
+        $host_ip_address_list = @gethostbynamel($host);
+
+        switch ($host) {
+            // Valid IPv4 loopback address
+            // FILTER_FLAG_NO_RES_RANGE contains `127.0.0.0/8`, but this can be potentially a valid host
+            // (especially for our PHPUnit tests), so we need to check this extra.
+            case filter_var($host, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4) and preg_match("/^127\./", $host):
+                break;
+            // Valid IPv6 loopback address
+            // FILTER_FLAG_NO_RES_RANGE contains `::1/128`, but this can be potentially a valid host
+            // (especially for our PHPUnit tests), so we need to check this extra.
+            case filter_var($host, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6) and (preg_match("/\:\:1$/", $host) or "0:0:0:0:0:0:0:1" == $host):
+                break;
+            // Valid IPv4 or IPv6 address without reserved CIDRs (e.g. APIPA)
+            case filter_var($host, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4|FILTER_FLAG_IPV6|FILTER_FLAG_NO_RES_RANGE):
+                break;
+            // Resolveable DNS hostname
+            // When an IP address gets resolved to itself, we do NOT accept this as resolveable host.
+            // This check is for resolving DNS hostnames.
+            case (is_array($host_ip_address_list) AND (count($host_ip_address_list) > 0) AND ($host_ip_address_list[0] != $host)):
```

Comment:
This can be simplified to
```php
case (($host_ip_address_list[0] ?? $host) != $host):
```

Reply:
The here used function [gethostbynamel()](https://www.php.net/manual/de/function.gethostbynamel.php) returns either an array (on success) or the boolean `false` (in case of could not be resolved).

That's the reason, why I explicitly check if `$host_ip_address_list` is an array or not. If it isn't an array and we try to access an index, we will get an error, which I wanted to avoid.

But yes, we could get rid of this `count()` condition. I would update it to this, if it's ok for you:

```php
case (is_array($host_ip_address_list) AND ($host_ip_address_list[0] != $host)):
```

Reply:
[Null coalesce](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce) (`??`) handles those scenarios all at once so we can slim down the code. :)

Reply:
Oh, cool. Fancy. I've changed it. :D

Reply:
All in the spirit of modernization!! 🎉